docs/manual/guides/install-test-versions.de.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Testversionen installieren"
+description: "Anleitung zur Installation von Entwicklungsversionen und Release Candidates."
+url: "anleitungen/testversionen-installieren"
+weight: 20
+---
+
+
+Das Entwicklerteam von Contao hält sich an einen festgelegten [Release-Plan][releasePlan]
+bei der Entwicklung neuer Versionen (bzw. dem Support bestehender). Vor jeder geplanten
+neuen Version gibt es einen gewissen Zeitraum in dem aktiv an dieser Version entwickelt
+wird. Neue Features werden hinzugefügt, bis ein gewisser Zeitpunkt erreicht ist.
+Danach werden die ersten »Release Candidates« veröffentlicht. Wenn während der Tests
+keine weiteren Probleme gefunden werden, wird die erste stabile Version veröffentlicht.
+
+Auch wenn Contao Unit Testing einsetzt können immer unvorhergesehen Probleme und
+Fehler auftauchen, nachdem neue Features hinzugefügt, oder bestehende Dinge
+geändert wurden. Daher müssen neue Versionen von tatsächlichen Benutzern getestet 
+werden. Als Open Source Software kann Contao dabei aber vom Engagement der Community 
+profitieren.
+
+Im Folgenden wird erklärt wie man neue Versionen von Contao selbst installieren
+und daher testen kann. Entweder durch die Installation eines Release Candidates,
+oder sogar einer aktuellen Entwicklerversion.
+
+Bei der Aktualisierung von Paketen über den Composer (entweder direkt auf der Kommandozeile
+oder über den Contao Manager) werden normalerweise immer nur _stabile_ Pakete installiert,
+zumindest mit den vorgegebenen Einstellungen der _Contao Managed Edition_. Daher
+muss die `composer.json` des Projektes entsprechend angepasst werden, damit auch
+Release Candidates oder Entwicklerversionen installiert werden. Nach der Änderung 
+der `composer.json` muss natürlich eine komplette Paketaktualisierung durchgeführt 
+werden. 
+
+
+## Release Candidates installieren
+
+Release Candidates nutzen eine spezifische Form von sogenannten »Release-Tags«.
+Der erste Release Candidate von Contao `4.9` heißt z. B. `4.9.0-RC1`. Mit Hilfe
+der »[Version Requirement Syntax][composerVersions]« von Composer können wir verlangen,
+dass solche Versionen ebenfalls installiert werden sollen, und nicht nur die stabilen
+Versionen. Diese Angabe erfolgt z. B. mit `4.9@RC`, und muss bei _allen_ Bundles
+des Contao Cores angewandt werden, _inklusive_ dem `contao/core-bundle` und dem
+`contao/installation-bundle`, welche normalerweise nicht direkt in der eigenen `composer.json`
+festgelegt werden.
+
+Hier ist ein komplettes Beispiel um die neueste Contao `4.9` Version installieren
+zu lassen, _inklusive_ den neuesten Release Candidates (wenn vorhanden):
+
+```json
+{
```

Comment:
I don't know but I find this way easier:

```json
{
    "require": {
        "contao/manager-bundle": "4.9.*"
    },
    "minimum-stability": "RC",
    "prefer-stable": true
}
```

It's easy to explain the `minimum-stability` and `prefer-stable` flags and all you have to do once the stable release is here is to remove those two lines and you're done.

Reply:
Yeah it's easier, but also causes a lot more processing time and memory usage within composer. I discussed this with @aschempp too and we agreed that the Contao Manager should also support this approach. @aschempp wdyt?

Reply:
Why would it use more processing time? It needs about 1.5 minutes in both cases anyway.

Reply:
We don't want composer to resolve _any_ RC version, we want it for a specific package. That's why we require them… It's also what you'll need to do in the Contao Manager, once we can install `core-bundle` and `installation-bundle`.

Reply:
You won‘t get any other RC version unless there‘s one of any other dependency **matching** our constraints which is highly unlikely to happen.
I‘ll choose my version over the explicit one any day. Even if it might potentially load an RC of another package, it‘s still way easier to use.

Reply:
> Why would it use more processing time? It needs about 1.5 minutes in both cases anyway.

|   | `minimum-stability` | `@RC` |
|---|---|---|
| Fresh install | 1923.78MiB, 248.31s | 1828.76MiB, 239.16s |
| Update | 1930.04MiB, 70.48s | 1834.38MiB, 69.41s |

It is currently a negligible difference, yeah.

Reply:
In any case, for installing the dev version you'd still use `4.X.x-dev` or `dev-master` anyway, because setting the `minimum-stability` to `dev` would be madness, wouldn't it? Therefore I think this approach is easier to grasp, as the changes necessary for installing an RC are equal to the ones for installing a dev version.

Reply:
It's much more convenient to add the `minimum-stability` and `prefer-stable` flags and remove them after the RC phase than touching each package in the `require` section.
+1 on @Toflar s proposal

Reply:
Or, I just document both ways :Þ

Reply:
In fact, because of `prefer-stable: true` it will only install RC versions **if not possible otherwise**. So in 99.99% of all the times you're doing this, you will not get any RC version of any other RC version except for Contao because we shouldn't release versions that require RC versions of dependencies. It's really highly unlikely.

Reply:
@contao/developers what do the others think? Which way should be the preferred and documented way?

Reply:
Definitely `4.9.x-dev`. Changing the minimum stability will affect **all** packages, not just the Contao ones!

Reply:
It will only affect them **if** there's not any stable release of those packages fulfilling the ruleset. That's **highly, highly** unlikely.

Reply:
> Definitely `4.9.x-dev`. Changing the minimum stability will affect **all** packages, not just the Contao ones!

Well yes, for the dev version of course. But what about the RC? There you can safely lower the minimum-stability to `RC`, without increasing the memory consumption or processing time of a composer update (too much) and without a big risk of installing release candidates of _other_ packages.

Reply:
IDK. I never have this case. 🤷‍♂

Reply:
I have now changed it to the `minimum-stability` version.